### ISSUE TITLE
Lavalink v2 roadmap

### ISSUE THREAD

**Issue by freyacodes** [lavalink-devs/Lavalink#33]
Lavalink v2 will be addressing issues about the robustness of Lavalink. By moving several actions to REST we can more easily assure that actions have occurred successfully, like playing a track or joining a voice channel.

# Server/protocol changes
- [ ] Make the REST server mandatory
- [ ] Create a "hello" OP which indicates the REST server base URL, as well as an authorization header
- [ ] Redo the docs
- [ ] Destroy method

## Replace these WS OPs with REST
- [ ] connect
- [ ] disconnect
- [ ] play
- [ ] stop
- [ ] pause
- [ ] seek
- [ ] volume

# JDA client changes
- [ ] Action queue similar to JDA's `RestAction` system
- [ ] Documentation changes where needed
- [ ] Remove deprecated methods

These changes aren't final and discussion is encouraged.

**Comment by freyacodes** [comment]
Over a month later we realize that this isn't the way to go. We have looked closer into what issues have been bugging us, and we have decided to move mainWS handling from LavaLink Server to the respective Discord library. What this means is that we threw a bunch of unnecessary (and broken) parts under the bus.

Draft implementation spec: https://github.com/Frederikam/Lavalink/commit/5b9b27fbf0cd64a078399306822751140e471a32

Notable changes:
* We no longer intend to do any of the initially proposed changes from December
* No OP 4 handling by Lavalink
* The removal of validation and connection checks
* The connect and disconnect ops are removed in favour of just having the respective Discord lib handle all that
* LavaLink Server will no longer attempt to reconnect (other than resuming)

This is all an effort to improve reliability while also making many things easier to implement.
